### ISSUE TITLE
eslint behaves differently on windows?

### ISSUE THREAD

**Issue by adrian-goe** [eslint/eslint#13118]
I put eslint into our project today.

the piplines are running. now some developers on windows have complained that eslint only throws errors with them.

If you take a closer look at them, they are indeed errors that should be found like 'no-magic-number'.

Interestingly, these are not found on mac or ubuntu(docker-image).

Is this a preference issue or is it a bug?

```
"eslint": "6.8.0",
"eslint-config-airbnb": "18.1.0",
"eslint-plugin-react": "7.19.0",
"eslint-plugin-react-hooks": "2.5.1",
```
`node:12.6.0`

**Comment by adrian-goe** [comment]
the eslint config

```
module.exports = {
    env: {
        browser: true,
        es6: true,
    },
    extends: [
        'plugin:react/recommended',
        'airbnb',
    ],
    globals: {
        Atomics: 'readonly',
        SharedArrayBuffer: 'readonly',
    },
    parser: '@typescript-eslint/parser',
    parserOptions: {
        ecmaFeatures: {
            jsx: true,
        },
        ecmaVersion: 2018,
        sourceType: 'module',
    },
    plugins: [
        'react',
        '@typescript-eslint',
    ],
    rules: {
        'no-console': 'error',
        'no-debugger': 'error',
        'no-extra-semi': 'error',
        'no-import-assign': 'error',
        'no-unreachable': 'error',
        'dot-location': ['error', 'object'],
        'dot-notation': ['error', {'allowKeywords': false}],
        'eqeqeq': ['error', 'always'],
        'no-alert': 'error',
        'no-empty-function': 'error',
        'no-iterator': 'error',
        'no-invalid-this': 'error',
        'no-magic-numbers': 'error',
        'no-multi-spaces': 'error',
        'no-script-url': 'error',
        'no-var': 'error',
        'no-unused-vars': 'error',
        'array-bracket-newline': ['error', {'multiline': true, 'minItems': 2}],
        'array-bracket-spacing': ['error', 'never'],
        'brace-style': 'error',
        'comma-spacing': ['error', {'before': false, 'after': true}],
        'camelcase': ['error', {allow: ['de_DE', 'en_US']}],
        'no-whitespace-before-property': 'error',
        'object-curly-newline': ['error', {'minProperties': 3}],
        "indent": ["error", 4]
        /*'no-use-before-define': 'error'*/
        /*'id-blacklist': ['error', 'data', 'err', 'e', 'cb', 'callback']*/
    },
};
```
```
eslint --version
v6.8.0
```

the expected result would be either on windows also no errors or on mac/linux the errors from on windows.

Since it looks like windows will find errors that are actually there, my expectation is that linux and mac will find the errors. But there are none.

Command to execute is
`"lint:eslint": "eslint src/**/*.js”,`

**Comment by mdjermanovic** [comment]
Thanks for the additional info!

Can you please also provide a code example where you see a difference between windows and mac/linux, along with the eslint output in both environments?

It certainly isn't an intentional behavior. Could it be a different eslint config?

[--print-config](https://eslint.org/docs/user-guide/command-line-interface#print-config) option might be helpful to find out that:

```bash
eslint --print-config path/to/file.js
```

**Comment by adrian-goe** [comment]
with path/to/file you mean the eslint or the file with the errors?

**Comment by mdjermanovic** [comment]
The file with the errors. 

It should print rules and other configuration options that apply to the given file.

Then we can see if the rules that were supposed to report errors (like `no-magic-numbers` you mentioned) are enabled for the given file.

**Comment by adrian-goe** [comment]
so for `no-magic-numbers` I found this config
```json
"no-magic-numbers": [
      "error",
      {
        "ignore": [],
        "ignoreArrayIndexes": true,
        "enforceConst": true,
        "detectObjects": false
      }
    ],

```

**Comment by adrian-goe** [comment]
and this is the log for windows
https://pastebin.com/D1pPdshR
same config for `no-magic-numbers`
```json
"no-magic-numbers": [
      "error",
      {
        "ignore": [],
        "ignoreArrayIndexes": true,
        "enforceConst": true,
        "detectObjects": false
      }
    ],
´´´

**Comment by mdjermanovic** [comment]
A code like

```js
if (response.status === 404 || (contentType != null && contentType.indexOf('javascript') === -1)) {
}
```

should have `no-magic-number` errors on `404` and `-1` with this configuration.

What happens when you run:

```bash
eslint ./src/serviceWorker.js
```

**Comment by adrian-goe** [comment]
now i get a lot of errors. So it works as intended

**Comment by adrian-goe** [comment]
I guess its solved with a small change
Before: `"lint:eslint": "eslint src/**/*.js”,`
After `"lint:eslint": "eslint './src/**/*.js'",`

**Comment by mdjermanovic** [comment]
Glad you solved the problem! It seems that the file(s) in question wasn't actually linted on mac/linux.

**Comment by kaicataldo** [comment]
To follow up for others who run into this issue, quoting the globs allows ESLint to do the expansion. Otherwise, the shell will handle it.

**Comment by kaicataldo** [comment]
Closing this issue since the question has been answered.
